### ISSUE TITLE
Angular Animation, two :enter / :leave animations require extra css?

### ISSUE THREAD

**Issue by borriej** [angular/angular#35030]
Hi,

**Goal**
I have a component that should animate two things:
 On Enter:
- Fade in,
- Slide-up a div

On Leave:
- Fade-out
- Slide-down the div

**Animation code**
```
    // the host / component should first fade in
    // we use HostBinding to bind this animation to the this (host) component
    trigger("fadeInState", [
      state("in", style({ opacity: 1 })),
      transition(":enter", [animate(".35s ease-out", style({ opacity: 1 }))]),
      transition(":leave", [animate(".35s ease-out", style({ opacity: 0 }))])
    ]),
    // in parallel, a div called '.activity' should animate 'up',
    // but its not animating, it just switches css classes, without transition...
    // >>> WHAT AM I DOING WRONG HERE ?
    trigger("slideUpState", [
      state("slide", style({ top: "20px" })),
      transition(":enter", [
        animate(".35s ease-in-out", style({ top: "20px" }))
      ]),
      transition(":leave", [animate(".35s ease-out", style({ top: "100vh" }))])
    ])
```

**Stackblitz**
https://stackblitz.com/edit/angular-kryqkn?file=src%2Fapp%2Fchild-view%2Fchild-view.component.ts

**The problem**

It seems the 2nd animation (slideUpState) is not tweening.. but it does switch states.

**workaround** 
I have added a extra css rule to the div (that should slide up) inside ```child-view.component.scss```
```
    transition: .2s ease-in-out all; // IT WORKS WITH TRANSTITON CSS RULE, but still weird
```

i find this very weird, because im using ```animate``` insde the ```transition``` triggers

**Comment by dario-piotrowicz** [comment]
Hi @borriej I think that the issue is in your animations setup, parent animations block the children's one so in your case the `fadeInState` kicks in and it blocks the `slideUpState` which doesn't get executed and leaves you with just the end result (the css transition you added then kicks in and animates the thing entirely on its own, totally independently from the angular animations).

what you need to do is in your `fadeInState` to `query` for your `slideUpState` and call `animateChild` to allow it to animate, you can also use a `group` to have both animations play at the same time, I've applied such tweaks to your stackblitz and all works as expected without your css transition: [stackblitz](https://stackblitz.com/edit/angular-b1z68y?file=src%2Fapp%2Fchild-view%2Fchild-view.component.ts)

For reference please check: [animateChild](https://angular.io/api/animations/animateChild#usage-notes)

I hope this helps :slightly_smiling_face:

**Comment by jessicajaniuk** [comment]
It appears this behaves as expected with some clarification from @dario-piotrowicz. If you still feel there is an issue, please provide further details in a new issue.
